### ISSUE TITLE
KVO: error when observing nested property within dictionary

### ISSUE THREAD

**Issue by triplef** [Microsoft/WinObjC#289]
We have the following setup:

1. Our test class "Test" has a property `a` of type ClassA
2. ClassA has a property `dict` of type NSMutableDictionary
3. That dictionary has a key-value pair of `b = <ClassB instance>`
4. ClassB has a primitive property `someProperty`

We are observing `a.dict.b.someProperty`. When triggering a KVO-notification by setting the property `a` we get the following error:

> NSInvalidArgumentException
> Reason: Class ClassA is not KVC compliant for key b.

Note that ClassA has no property named `b`, but instead `b` is the key used in the dictionary. Also, the exception occurs *after* `-observeValueForKeyPath:` is called.

Calling `-[Test test]` in the following code reproduces the issue:

```objc
@interface ClassA : NSObject
@property (strong, nonatomic) NSMutableDictionary *dict;
@end

@implementation ClassA
@end

@interface ClassB : NSObject
@property (assign, nonatomic) float someProperty;
@end

@implementation ClassB
@end

@interface Test : NSObject
@property (strong, nonatomic) ClassA *a;
@end

@implementation Test

- (void)test
{
	ClassB *b = [[ClassB alloc] init];
	b.someProperty = 1.23;

	NSMutableDictionary *dict = [NSMutableDictionary dictionary];
	[dict setObject:b forKey:@"b"];

	ClassA *a = [[ClassA alloc] init];
	a.dict = dict;

	self.a = a;

	[self addObserver:self forKeyPath:@"a.dict.b.someProperty" options:0 context:nil];

	self.a = a;	// triggers KVO notification
}

- (void)observeValueForKeyPath:(NSString *)keyPath ofObject:(id)object change:(NSDictionary *)change context:(void *)context
{
}

@end
```

**Comment by DHowett-MSFT** [comment]
On it. Testing out a fix locally. Thanks for all the test cases!

**Comment by rajsesh-msft** [comment]
Available in release 0.1.0.151222
